libsolidity/ast/Types.cpp

```diff
@@ -566,12 +575,30 @@ BoolResult IntegerType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 
 BoolResult IntegerType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 {
-	return _convertTo.category() == category() ||
-		_convertTo.category() == Category::Address ||
-		_convertTo.category() == Category::Contract ||
-		_convertTo.category() == Category::Enum ||
-		(_convertTo.category() == Category::FixedBytes && numBits() == dynamic_cast<FixedBytesType const&>(_convertTo).numBytes() * 8) ||
-		_convertTo.category() == Category::FixedPoint;
+	if (auto integerType = dynamic_cast<IntegerType const*>(&_convertTo))
+	{
+		if (
+			(numBits() == integerType->numBits()) ||
+			(isSigned() == integerType->isSigned())
+		)
+			return true;
+	}
+	else if (_convertTo.category() == Category::Address)
+	{
+		if (!isSigned() && numBits() == 160)
+			return true;
+	}
+	else if (auto fixedBytesType = dynamic_cast<FixedBytesType const*>(&_convertTo))
+	{
+		if (!isSigned() && (numBits() == fixedBytesType->numBytes() * 8))
+			return true;
+	}
+	else if (dynamic_cast<EnumType const*>(&_convertTo))
+		return true;
```

Comment:
So no checking the actual size of enums? Not sure what was decided

Reply:
We are not. chris mentioned in the chat that this is undecided. What do you think we should do?

Reply:
Personally I think they should have the same rules using the uint type the enum gets converted to (smallest uint that fits). However, apparently we're going to make them uint8 always, so it might make sense to already assume enum is uint8.

Reply:
I agree with this. However, @chriseth and @cameel were not very keen about this change.

Reply:
At least we should not allow signed integer to enum conversion. Having to write `MyEnum(uint8(x))` is a bit cumbersome and I don't think it provides a benefit. Do at least literals work?

Reply:
Literals would work if the literal fits in the enum.